### PR TITLE
update on SetupTDCS; (1) contact surface extraction (2) point electrodes

### PR DESCRIPTION
**Pull request by moritzdannhauer**
I added a new checkbox. 

One GUI problem:
It seems that the checkbox status (if its checked or not) gets saved in the network file
but does not reflect the GUI after reloading.

@dcwhite Could you have a look at this.

Thanks!

### CODE REVIEW

File: src/Core/Algorithms/BrainStimulator/SetupRHSforTDCSandTMSAlgorithm.h [hunk 1]
```diff
@@ -87,8 +88,8 @@ namespace BrainStimulator {
...
     static const double electode_current_summation_bound;
-    SCIRun::Core::Datatypes::DenseMatrixHandle create_rhs(FieldHandle mesh, const std::vector<Variable>& elcs, int num_of_elc) const;
-    boost::tuple<Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, FieldHandle, std::vector<double>> create_lhs(FieldHandle mesh, const std::vector<Variable>& impelc, FieldHandle scalp_tri_surf, FieldHandle elc_tri_surf, SCIRun::Core::Datatypes::DenseMatrixHandle elc_sponge_location) const;
+    SCIRun::Core::Datatypes::DenseMatrixHandle create_rhs(FieldHandle mesh, FieldHandle elc_tri_surf, const std::vector<Variable>& elcs, int num_of_elc) const;
+    boost::tuple<Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, Datatypes::DenseMatrixHandle, FieldHandle, std::vector<double>> create_lhs(FieldHandle mesh, const std::vector<Variable>& impelc, FieldHandle scalp_tri_surf, FieldHandle elc_tri_surf, SCIRun::Core::Datatypes::DenseMatrixHandle elc_sponge_location) const;
```
Comment (dcwhite) [inline src/Core/Algorithms/BrainStimulator/SetupRHSforTDCSandTMSAlgorithm.h]: I suggest using a typedef for this giant tuple.

File: src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc [hunk 2]
```diff
@@ -63,6 +63,8 @@ void SetupTDCS::setStateDefaults()
   setStateDoubleFromAlgo(Parameters::normal_dot_product_bound);
   setStateDoubleFromAlgo(Parameters::pointdistancebound);
   setStateIntFromAlgo(Parameters::number_of_electrodes);
+  setStateBoolFromAlgo(Parameters::GetContactSurface);
+  state->setValue(Parameters::GetContactSurface, false);
```
Comment (dcwhite) [inline src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc]: @moritzdannhauer why this line? you just overwrote the value from the network file.

File: src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc [hunk 3]
```diff
@@ -84,13 +86,13 @@ void SetupTDCS::execute()
...
-    //state->setValue(Parameters::normal_dot_product_bound, get_state()->getValue(Parameters::normal_dot_product_bound).toDouble());
-    //state->setValue(Parameters::pointdistancebound, get_state()->getValue(Parameters::pointdistancebound).toDouble());
-    
+   
     setAlgoIntFromState(Parameters::refnode);   
     setAlgoDoubleFromState(Parameters::normal_dot_product_bound);
     setAlgoDoubleFromState(Parameters::pointdistancebound);
+    setAlgoBoolFromState(Parameters::GetContactSurface);
+    
+    algo().set(Parameters::GetContactSurface, state->getValue(Parameters::GetContactSurface).toBool());
```
Comment (dcwhite) [inline src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc]: @moritzdannhauer Don't need this line, line 93 does what you need.

File: src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc (hunk 2 above)
Comment (moritzdannhauer) [inline src/Modules/BrainStimulator/SetupRHSforTDCSandTMS.cc]: Yes, I had problems with that one. The GetContactSurface boolean gets saved in the network file but
after reloading it does not show up as being "checked" in the GUI. Can you have a look at that?
Thanks!

File: src/Core/Algorithms/BrainStimulator/SetupRHSforTDCSandTMSAlgorithm.h (hunk 1 above)
Comment (moritzdannhauer) [inline src/Core/Algorithms/BrainStimulator/SetupRHSforTDCSandTMSAlgorithm.h]: I tried something to define all output parameters as a new type but I could not get it to compile.

**Comment by dcwhite** [comment]
Once again, you're missing the call to pull_newVersionToReplaceOld() inside pull() in the dialog class. You still need that.

**Comment by dcwhite** [comment]
@moritzdannhauer Once again, you're missing the call to pull_newVersionToReplaceOld() inside pull() in the dialog class. You still need that.

**Comment by dcwhite** [comment]
@moritzdannhauer line 111

**Comment by moritzdannhauer** [comment]
Done! Thanks for the heads up on the "pull_newVersionToReplaceOld()" issue. 
I dont get emails when you comment on pull requests - can we enable that?

**Comment by dcwhite** [comment]
Check your github notification settings, accessed by the gear at the top right of any page.

**Comment by dcwhite** [comment]
Testing comment.

**Pull request merged**